app/models/broadcast_message.py

```
@@ -74,7 +78,12 @@ def from_id(cls, broadcast_message_id, *, service_id):
 
     @property
     def areas(self):
-        return self.get_areas(areas=self._dict['areas'])
+        return self.get_areas(
+            areas=self._dict['areas']
+        ) or CustomBroadcastAreas(
+            areas=self._dict['areas'],
+            polygons=self._dict['simple_polygons'],
+        )
```

can we do a sanity check at this point that the length of the list returned is the same as the size of the `self._dict`? that'll catch the (admittedly, very edge casey) scenario where someone's included one place we recognise and one place we don't recognise. not sure what we should do in that situation, but i'd rather we raise an exception than just silently discard the unrecognised areas

---

Good idea, added a check in 926ada2f2